estuary_updater/handlers/base.py

```diff
@@ -89,42 +89,52 @@ def is_container_build(self, build_info):
         else:
             return False
 
-    def get_or_create_build(self, identifier, original_nvr=None, force_container_label=False):
+    def get_or_create_build(self, identifier, build_info=None, original_nvr=None,
+                            force_container_label=False):
         """
         Get a Koji build from Neo4j, or create it if it does not exist in Neo4j.
 
         :param str/int identifier: an NVR (str) or build ID (int)
+        :param dict build_info: all info needed to create a build, else get from a Koji session
         :kwarg str original_nvr: original_nvr property for the ContainerKojiBuild
         :kwarg bool force_container_label: when true, this skips the check to see if the build is a
         container and just creates the build with the ContainerKojiBuild label
         :rtype: KojiBuild
         :return: the Koji Build retrieved or created from Neo4j
         """
-        try:
-            koji_build_info = self.koji_session.getBuild(identifier, strict=True)
-        except Exception:
-            log.error('Failed to get brew build using the identifier {0}'.format(identifier))
-            raise
+        if build_info is None:
+            try:
+                build_info = self.koji_session.getBuild(identifier, strict=True)
+            except Exception:
+                log.error('Failed to get brew build using the identifier {0}'.format(identifier))
+                raise
 
         build_params = {
-            'completion_time': datetime.fromtimestamp(int(koji_build_info['completion_ts'])),
-            'creation_time': datetime.fromtimestamp(int(koji_build_info['creation_ts'])),
-            'epoch': koji_build_info['epoch'],
-            'extra': json.dumps(koji_build_info['extra']),
-            'id_': str(koji_build_info['id']),
-            'name': koji_build_info['package_name'],
-            'release': koji_build_info['release'],
-            'start_time': datetime.fromtimestamp(int(koji_build_info['start_ts'])),
-            'state': koji_build_info['state'],
-            'version': koji_build_info['version']
+            'completion_time': None,
```

Comment:
Instead of making this `None`, you could just delete these line since you set it later on. Same with the other two.